packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -554,9 +553,10 @@ self.addEventListener("fetch", (event) => {
   if (event.request.url == origin || event.request.url.startsWith(origin + '/#') || key == '') {
     key = '/';
   }
-  // If the URL is not the RESOURCE list, skip the cache.
+  // If the URL is not the RESOURCE list then return to signal that the
+  // browser should take over.
   if (!RESOURCES[key]) {
-    return event.respondWith(fetch(event.request));
+    return;
   }
```

Comment:
Ah, I was understanding "remove main.dart.js bypass" wrong. This lets the service worker cache the main.dart.js again, by not overriding its name with a hash.

(I was about to ask: if we want the serviceworker to bypass main.dart.js, shouldn't we also remove it from the RESOURCES map? :P)

Reply:
Yes, exactly - now that I realize I was COMPLETE WRONG about the default behavior of sw, everything makes more sense!

But we also bypass the worker for http requests that won't go to the cache, since I guess there is overhead for forwarding the requests